pkg/gitops/gitops_test.go

```diff
@@ -0,0 +1,194 @@
+package gitops
+
+import (
+	"context"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/mock"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"path/filepath"
+)
+
+type mockCloner struct {
+	mock.Mock
+}
+
+func (m *mockCloner) CloneRepo(cloneDirPrefix string, branch string, gitURL string) (string, error) {
+	args := m.Called(cloneDirPrefix, branch, gitURL)
+	return args.String(0), args.Error(1)
+}
+
+var _ = Describe("gitops profile", func() {
+
+	var (
+		gitCloner *mockCloner
+		memFs     afero.Fs
+		io        afero.Afero
+		testDir   string
+		profile   *Profile
+		outputDir string
+		processor fileprocessor.FileProcessor
+	)
+
+	Context("generating a profile", func() {
+
+		BeforeEach(func() {
+			// In memory filesystem for the tests
+			memFs = afero.NewMemMapFs()
+			io = afero.Afero{Fs: memFs}
+
+			// Create test data files instead of cloning
+			testDir, _ = io.TempDir("", "test-dir-")
+			createTestFiles(testDir, memFs)
+
+			// mock git clone
+			gitCloner = new(mockCloner)
+			gitCloner.On("CloneRepo", mock.Anything, mock.Anything, mock.Anything).Return(testDir, nil)
+
+			// output path
+			outputDir, _ = io.TempDir("", "test-output-dir-")
+
+			processor = &fileprocessor.GoTemplateProcessor{
+				Params: fileprocessor.TemplateParameters{ClusterName: "test-cluster"},
+			}
+			profile = &Profile{
+				Path: outputDir,
+				GitOpts: GitOptions{
+					Branch: "master",
+					URL:    "git@github.com:someorg/test-gitops-repo.git",
+				},
+				IO:        io,
+				Fs:        memFs,
+				GitCloner: gitCloner,
+				Processor: processor,
+			}
+		})
+
+		AfterEach(func() {
+			io.RemoveAll(testDir)
+			io.RemoveAll(outputDir)
+		})
+
+		It("processes go templates and writes them to the output directory", func() {
+			err := profile.Generate(context.Background())
+
+			Expect(err).ToNot(HaveOccurred())
+			template1, err := io.ReadFile(filepath.Join(outputDir, "a/good-template1.yaml"))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(template1).To(Equal([]byte("cluster: test-cluster")))
```

Comment:
I guess less important for single line, but you could use `MatchYAML` here also.